manufacturers/prob_match_manufacturers.R

```diff
@@ -0,0 +1,186 @@
+
+goodmatches <- rbind(goodmatches, a82_83[c(38,8),])
+goodmatches
+
+
+# ========== Create Key Values ============ ####
+
+#Looks like this is a good stopping point. So now we have our list of good matches. (Almost 200!)
+#To set a key list, we should take this list and make a unique list of the codr values, because they are cleaned and known to be unique. 
+#Each of those gets a key value, which will be used dict style to apply to the lob values.
+companies_list <- list(unique(as.character(goodmatches$c_names)), as.numeric(as.character(paste0(seq(1:length(unique(as.character(goodmatches$c_names)))), "1001"))))
+
+#Since R doesn't have dicts, you can just use it this way, making reference calls, if you want to:
+companies_list[[1]][1]
+companies_list[[2]][1]
+
+#Just for convenience I'm also making a df version that I'll use to do merges:
+companies_table <- as.data.frame(cbind(unlist(companies_list[[1]]), unlist(companies_list[[2]])))
+
+
+# ======= Apply the key ============ #
+
+#easy!
+keyed_companies <- merge(goodmatches[,c("c_names", "l_names")],companies_table, by.x="c_names", by.y="V1")
+colnames(keyed_companies)[3] <- "company_key"
+keyed_companies$company_key <- as.numeric(as.character(keyed_companies$company_key))
+
+#To get back to the beginning, then, let's return to our original datasets and key those.
+#Remember, a lot of the records won't get a key in this because they didn't match.
+companies_drugs_keyed <- merge(companies_drugs, keyed_companies[,c("c_names", "company_key")], by.x="LABELER.NAME", by.y="c_names", all=T)
+lobbying$client <- toupper(lobbying$client)
+lobbying_keyed <- merge(lobbying, keyed_companies[,c("l_names", "company_key")], by.x="client", by.y="l_names", all=T)
+
+
+# ============== Keys for nonmatched ===================== #
+
+#Now it's time to add keys for those in each file that didn't get records. 
+
+lobby_list <- unique(lobbying_keyed$client[is.na(lobbying_keyed$company_key)])
+key_list <- paste0(seq(117,5000), "1002") #using 1002 as the suffix to identify lobbying nonmatches.
+lobby_list_table <- as.data.frame(cbind(lobby_list, key_list[1:length(lobby_list)]))
+lobbying_keyed <- merge(lobbying_keyed, lobby_list_table, by.x=c("client"), by.y=c("lobby_list"), all=T)
+lobbying_keyed$company_key <- ifelse(is.na(lobbying_keyed$company_key), as.numeric(as.character(lobbying_keyed$V2)), lobbying_keyed$company_key)
+
+cd_list <- unique(companies_drugs_keyed$LABELER.NAME[is.na(companies_drugs_keyed$company_key)])
+key_list <- paste0(seq(1240,5000), "1003") #using 1003 as the suffix to identify company-drug nonmatches.
+cd_list_table <- as.data.frame(cbind(cd_list, key_list[1:length(cd_list)]))
+companies_drugs_keyed <- merge(companies_drugs_keyed, cd_list_table, by.x=c("LABELER.NAME"), by.y=c("cd_list"), all=T)
+companies_drugs_keyed$company_key <- ifelse(is.na(companies_drugs_keyed$company_key), as.numeric(as.character(companies_drugs_keyed$V2)), companies_drugs_keyed$company_key)
+
```

Comment:
Ah, that is an area I could use a reference for. It doesn't actually matter, because of the way I'm suffixing the keys, but when I was working I made the decision about key suffixes on the fly. The values came from the endpoint where the sequence had left off, fwiw. I'll adjust that, this makes sense.